### DISCUSSION TITLE
Getting ValueError: Can only tuple-index with a MultiIndex

### DISCUSSION THREAD

**Discussion by adiv5** [scikit-multilearn/scikit-multilearn#283] [Q&A] [answered]
I am trying to stratify my multi-label data, `total` is all my data. contains 20 columns, 1st column is text(X) and rest of 19 cols are labels( each col represent a class, if present for an example,set to 1 else set to 0). `total` is a csv file if this info is needed
```
from skmultilearn.model_selection import iterative_train_test_split
X_train, y_train, X_test, y_test = iterative_train_test_split(total.iloc[:,0], total.iloc[:,1:], test_size = 0.5)
```
i am getting the following Error:
```
ValueError: Can only tuple-index with a MultiIndex
```
Here is the traceback:
```
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-9-98829124e697> in <module>
      1 from skmultilearn.model_selection import iterative_train_test_split
----> 2 X_train, y_train, X_test, y_test = iterative_train_test_split(total.iloc[:,0], total.iloc[:,1:], test_size = 0.5)

~/virtualenvs/anaconda3/envs/tf1/lib/python3.6/site-packages/skmultilearn/model_selection/iterative_stratification.py in iterative_train_test_split(X, y, test_size)
     93     train_indexes, test_indexes = next(stratifier.split(X, y))
     94 
---> 95     X_train, y_train = X[train_indexes, :], y[train_indexes, :]
     96     X_test, y_test = X[test_indexes, :], y[test_indexes, :]
     97 

~/virtualenvs/anaconda3/envs/tf1/lib/python3.6/site-packages/pandas/core/series.py in __getitem__(self, key)
   1111             key = check_bool_indexer(self.index, key)
   1112 
-> 1113         return self._get_with(key)
   1114 
   1115     def _get_with(self, key):

~/virtualenvs/anaconda3/envs/tf1/lib/python3.6/site-packages/pandas/core/series.py in _get_with(self, key)
   1125         elif isinstance(key, tuple):
   1126             try:
-> 1127                 return self._get_values_tuple(key)
   1128             except Exception:
   1129                 if len(key) == 1:

~/virtualenvs/anaconda3/envs/tf1/lib/python3.6/site-packages/pandas/core/series.py in _get_values_tuple(self, key)
   1170 
   1171         if not isinstance(self.index, MultiIndex):
-> 1172             raise ValueError("Can only tuple-index with a MultiIndex")
   1173 
   1174         # If key is contained, would have returned by now

ValueError: Can only tuple-index with a MultiIndex
```
Thanks in advance.

**Comment by jomedinag-zz** [comment]
I get the same error. No answer yet.

**Comment by eosulliv** [comment]
Has anyone found an answer or a solution to this yet? I have just encountered the same problem, but thus far my attempts to solve it have proven fruitless.

**Comment by eosulliv** [comment]
After some further investigation and going through the example found on the scikit-mltilearn website [here](http://scikit.ml/stratification.html), I think the issue may be with how the X and Y data are passed. I encountered this error when passing in the data from a pandas dataframe:
`iterative_train_test_split(df['A'], df['B', 'C', 'D'], test_size=0.5)`
Converting from pandas dataframe to matrices prior to passing the data into the function worked for me.

**Comment by vitormm44** [comment]
> After some further investigation and going through the example found on the scikit-mltilearn website [here](http://scikit.ml/stratification.html), I think the issue may be with how the X and Y data are passed. I encountered this error when passing in the data from a pandas dataframe:
> `iterative_train_test_split(df['A'], df['B', 'C', 'D'], test_size=0.5)`
> Converting from pandas dataframe to matrices prior to passing the data into the function worked for me.

Can you be more specific? I've tried to convert my data to matrices, but it didn't work out. What type of matrices did you use and how did you do this conversion?

**Comment by eosulliv** [comment]
I converted the data to matrices using the numpy `asmatrix` function.
```
import numpy as np
X_train, y_train, X_test, y_test = iterative_train_test_split(np.asmatrix(df['A']), np.asmatrix(df['B', 'C', 'D']), test_size=0.5)
```

I hope this helps.

**Comment by charlottjakob** [comment, marked as answer]
I had the same problem I could fix it by reshaping my data. 
Both inputs X and y had to be a 2 dimensional matrix or array and the first dimension needed to have the same length. 
E.g. if X is only one column select is as dataframe or reshape the array

Methode 1 select as dataframe:
X = df[ ['x_col']].values
y = df[ ['y_col1', 'y_col2‘]].values

Methode 2 reshape array:
X = df[ ['x_col']].values.reshape((len(df), -1))
y = df[ ['y_col1', 'y_col2‘]].values

X_train, y_train, X_test, y_test = iterative_train_test_split(X, y, test_size=0.5)

I hope it helps :)
